### PR TITLE
docs: clarify xds docs with warming information

### PR DESCRIPTION
**Pull request by ramaraochavali**
Signed-off-by: Rama Chavali <rama.rao@salesforce.com>
Description: As a follow-up to #6151 , this PR updates xds docs.
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@mattklein123  @htuch  PTAL

File: api/XDS_PROTOCOL.md [hunk 1]
```diff
@@ -209,6 +211,20 @@ multiple `DiscoveryRequests` at a version until a new version is ready.
...
 expect a `DiscoveryResponse` for every `DiscoveryRequest` it issues.
 
+### Resource warming
+
+[`Clusters`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/cluster_manager.html#cluster-warming) 
+and [`Listeners`](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/lds#config-listeners-lds)
+go through  `warming` before they can serve requests. This process happens both during 
+[`Envoy initialization`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/init.html#initialization) 
+and when the `Cluster` or `Listener` is updated. Warming of `Cluster` is completed only when a 
+`ClusterLoadAssignment` response is supplied by management server. Similarly warming of `Listener`
```
Comment (venilnoronha) [inline api/XDS_PROTOCOL.md]: nit: s/Similarly/Similarly,

File: api/XDS_PROTOCOL.md [hunk 2]
```diff
@@ -209,6 +211,20 @@ multiple `DiscoveryRequests` at a version until a new version is ready.
...
+is completed only when a `RouteConfiguration`  is suppled by management server if the listener
```
Comment (venilnoronha) [inline api/XDS_PROTOCOL.md]: nit: remove extra space after \`RouteConfiguration\`

File: api/XDS_PROTOCOL.md [hunk 3]
```diff
@@ -209,6 +211,20 @@ multiple `DiscoveryRequests` at a version until a new version is ready.
...
+### Resource warming
+
+[`Clusters`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/cluster_manager.html#cluster-warming) 
+and [`Listeners`](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/lds#config-listeners-lds)
+go through  `warming` before they can serve requests. This process happens both during 
+[`Envoy initialization`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/init.html#initialization) 
+and when the `Cluster` or `Listener` is updated. Warming of `Cluster` is completed only when a 
+`ClusterLoadAssignment` response is supplied by management server. Similarly warming of `Listener`
+is completed only when a `RouteConfiguration`  is suppled by management server if the listener 
+refers to an RDS configuration. Management Server is expected to provide the EDS/RDS updates during
```
Comment (venilnoronha) [inline api/XDS_PROTOCOL.md]: nit: s/Server/server

File: api/XDS_PROTOCOL.md [hunk 4]
```diff
@@ -209,6 +211,20 @@ multiple `DiscoveryRequests` at a version until a new version is ready.
...
+warming. If management server does not provide EDS/RDS information, Envoy will not initialize
+it self during initialization phase and the updates sent via CDS/LDS will not take effect till
```
Comment (venilnoronha) [inline api/XDS_PROTOCOL.md]: s/it self/itself

File: api/XDS_PROTOCOL.md [hunk 5]
```diff
@@ -209,6 +211,20 @@ multiple `DiscoveryRequests` at a version until a new version is ready.
...
+[`Clusters`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/cluster_manager.html#cluster-warming) 
+and [`Listeners`](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/lds#config-listeners-lds)
+go through  `warming` before they can serve requests. This process happens both during 
+[`Envoy initialization`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/init.html#initialization) 
+and when the `Cluster` or `Listener` is updated. Warming of `Cluster` is completed only when a 
+`ClusterLoadAssignment` response is supplied by management server. Similarly, warming of `Listener`
+is completed only when a `RouteConfiguration` is suppled by management server if the listener 
+refers to an RDS configuration. Management server is expected to provide the EDS/RDS updates during
+warming. If management server does not provide EDS/RDS responses, Envoy will not initialize
+itself during initialization phase and the updates sent via CDS/LDS will not take effect till
```
Comment (venilnoronha) [inline api/XDS_PROTOCOL.md]: Drop "during initialization phase"?

Comment (ramaraochavali) [reply api/XDS_PROTOCOL.md]: I was in two minds when I wrote this - but decided to add this for more clarity. LMK if you strong feel about removing, I can remove.

Comment (venilnoronha) [reply api/XDS_PROTOCOL.md]: Up to you. I don't feel too strongly.

File: api/XDS_PROTOCOL.md [hunk 6]
```diff
@@ -209,6 +211,20 @@ multiple `DiscoveryRequests` at a version until a new version is ready.
...
 
+### Resource warming
+
+[`Clusters`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/cluster_manager.html#cluster-warming) 
+and [`Listeners`](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/lds#config-listeners-lds)
+go through  `warming` before they can serve requests. This process happens both during 
+[`Envoy initialization`](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/init.html#initialization) 
+and when the `Cluster` or `Listener` is updated. Warming of `Cluster` is completed only when a 
+`ClusterLoadAssignment` response is supplied by management server. Similarly, warming of `Listener`
+is completed only when a `RouteConfiguration` is suppled by management server if the listener
```
Comment (htuch) [inline api/XDS_PROTOCOL.md]: Nit: s/suppled/supplied/

File: api/XDS_PROTOCOL.md (hunk 5 above)
Comment (htuch) [inline api/XDS_PROTOCOL.md]: Nit: "the initialization phase", s/till/untill/

**Pull request merged**